src/Nethermind/Nethermind.Init/Steps/InitializePrecompiles.cs

```diff
@@ -13,10 +13,12 @@ namespace Nethermind.Init.Steps;
 public class InitializePrecompiles : IStep
 {
     private readonly INethermindApi _api;
+    private readonly IInitConfig _initConfig;
 
     public InitializePrecompiles(INethermindApi api)
     {
         _api = api;
+        _initConfig = api.Config<IInitConfig>();
```

Comment:
Very small optimization: _initConfig can be retrieved lazily inside Execute's if clause. Actually the logger can be materialized in exception clause also.